datadog/dogstatsd/base.py

```diff
@@ -265,6 +282,9 @@ def _send_to_server(self, packet):
         try:
             # If set, use socket directly
             (self.socket or self.get_socket()).send(packet.encode(self.encoding))
+        except socket.timeout:
+            # dogstatsd is overflowing, drop the packets (mimicks the UDP behaviour)
+            return
```

Comment:
The python socket implementation [implements non-blocking send as a zero-timeout send](https://docs.python.org/2/library/socket.html#socket.socket.setblocking) and sends a `socket.timeout` exception if the write does not return immediately (the queue, managed by the kernel, is full), I chose to silently drop the packet, mirroring what UDP would do. With dogstatsd6's goroutine-based intake, this should only happen if the CPU is saturated.

If needed, the socket queue length is configurable via sysctl net.unix.max_dgram_qlen (10 on my test machines).